waagent

```diff
@@ -2551,6 +2564,31 @@ class FreeBSDDistro(AbstractDistro):
         SetFileContents(location+"/ovf-env.xml", ovfxml)
         return retcode,out
 
+    def mkfile(self,filename,nbytes):
+        """
+        Create a non-sparse file of that size. Replace existing file. BSD-compatible dd version.
+        
+        For sizes < 64M, a single-pass dd is used. Otherwise use two-pass dd.
+        """
+        max_blocksize = 64 * 1024 ** 2
+        dd_cmd = "dd if=/dev/zero bs={bs} count={n} conv=notrunc of={fn}"
+        
+        if not type(nbytes) is int:
+           nbytes = int(nbytes)
+        
+        if os.path.isfile(filename):
+            os.remove(filename)
+        
+        fn_sh = ShellEscape(filename)
```

Comment:
You don't need the power of the shell, so don't set yourself up for that heartache. Build this as an array of strings. Since you need to do it multiple times, build a function that takes blocksize, count, and filename as parameters and returns the array of strings to pass to Run().